test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -0,0 +1,258 @@
+        assertTrue(defaultAddressBook.containsTag(tagPrizeWinner));
+        
+        assertTrue(isTagObjectInAddressBookList(tagEconomist, defaultAddressBook));
+        assertTrue(isTagObjectInAddressBookList(tagPrizeWinner, defaultAddressBook));
+    }
+
+    @Test
+    public void addPerson_personAlreadyInList_throwsDuplicatePersonException() throws Exception {
+        thrown.expect(DuplicatePersonException.class);
+        defaultAddressBook.addPerson(aliceBetsy);
+    }
+    
+    @Test
+    public void addPerson_personAlreadyInListButHasTagNotInList_tagNotAdded() throws Exception {
+        aliceBetsy.setTags(new UniqueTagList(tagMathematician, tagPrizeWinner));
+        
+        try {
+            defaultAddressBook.addPerson(aliceBetsy);
+        } catch (DuplicatePersonException e) {
+            assertFalse(defaultAddressBook.containsTag(tagPrizeWinner));
+            return;
+        }
+        
+        fail("Expected DuplicatePersonException was not caught.");
+    }
+
+    @Test
+    public void addTag_tagNotInList_addsNormally() throws Exception {
+        assertFalse(defaultAddressBook.containsTag(tagEconomist));
+        defaultAddressBook.addTag(tagEconomist);
+        
+        UniqueTagList expectedTagsAfterAddition = new UniqueTagList(tagMathematician, tagScientist, tagEconomist);
+        assertTrue(isIdentical(expectedTagsAfterAddition, defaultAddressBook.getAllTags()));
+    }
+
+    @Test
+    public void addTag_tagAlreadyInList_throwsDuplicateTagException() throws Exception {
+        thrown.expect(DuplicateTagException.class);
+        defaultAddressBook.addTag(tagMathematician);
+    }
+
+    @Test
+    public void containsPerson() throws Exception {
+        UniquePersonList personsWhoShouldBeIn = new UniquePersonList(aliceBetsy, bobChaplin);
+        UniquePersonList personsWhoShouldNotBeIn = new UniquePersonList(charlieDouglas, davidElliot);
+        
+        for (Person personWhoShouldBeIn : personsWhoShouldBeIn) {
+            assertTrue(defaultAddressBook.containsPerson(personWhoShouldBeIn));
+        }
+        for (Person personWhoShouldNotBeIn : personsWhoShouldNotBeIn) {
+            assertFalse(defaultAddressBook.containsPerson(personWhoShouldNotBeIn));
+        }
+        
+        UniquePersonList allPersons = new UniquePersonList(aliceBetsy, bobChaplin, charlieDouglas, davidElliot);
+        
+        for (Person person : allPersons) {
+            assertFalse(emptyAddressBook.containsPerson(person));
+        }
+    }
+
+    @Test
+    public void containsTag() throws Exception {
+        assertTrue(defaultAddressBook.containsTag(tagScientist));
+        assertFalse(defaultAddressBook.containsTag(tagEconomist));
+        
+        UniqueTagList allTags = new UniqueTagList(tagPrizeWinner, tagScientist, tagMathematician, tagEconomist);
+        
+        for (Tag tag : allTags) {
+            assertFalse(emptyAddressBook.containsTag(tag));
+        }
+    }
```

Comment:
Why the above two methods have different structures? They are testing similar things right?